### ISSUE TITLE
Add some Opt-ins, pls: Use class properties instead of archaic incorrect bindings in constructor, and remove initClass

### ISSUE THREAD

**Issue by a-x-** [decaffeinate/decaffeinate#1358]
```coffee
class Search extends React.PureComponent
  @propTypes:
    onChange: PropTypes.func.isRequired

  constructor: (props) ->
    super props
    @inputRef = React.createRef()

  renderInput: =>
    <input/>
```

[https://decaffeinate-project.org/repl...](https://decaffeinate-project.org/repl/#?useCS2=true&useJSModules=true&loose=true&evaluate=true&stage=full&code=class%20Search%20extends%20React.PureComponent%0A%20%20%40propTypes%3A%0A%20%20%20%20onChange%3A%20PropTypes.func.isRequired%0A%0A%20%20constructor%3A%20(props)%20-%3E%0A%20%20%20%20super%20props%0A%20%20%20%20%40inputRef%20%3D%20React.createRef()%0A%0A%20%20renderInput%3A%20%3D%3E%0A%20%20%20%20%3Cinput%2F%3E%0A)
I get this error:

Would be much better:
```
class Search extends TwoColumnsViewSearch {
    static propTypes = {
      onChange: PropTypes.func.isRequired,
    };

  constructor(props) {
    super(props);
    this.inputRef = React.createRef();
  }

  renderInput = () => {
    return <input/>;
```

**Comment by a-x-** [comment]
can I contribute it manually quickly?

**Comment by a-x-** [comment]
I could not patch decaffeinate (I spent 3 hours and I had not got behind the broken sourcemaps and weird exceptions), so I wrote some AST-based post-processors:
- **rm initClass** ([gist](https://gist.github.com/a-x-/adc99a6c115a2df67adade6250fca81a))
- **binds to props** _WIP_

**Comment by eventualbuddha** [comment]
I agree. This would be better given a `--use-class-properties` option.

**Comment by akx** [comment]
You can use [`npx react-codemod manual-bind-to-arrow app.js`](https://github.com/reactjs/react-codemod) on the decaffeinated code for the time being :)
